### PR TITLE
HTTP Beacon API: `/eth/v1/beacon/headers`

### PR DESCRIPTION
**Pull request by anukul**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

HTTP implementation of https://ethereum.github.io/beacon-APIs/#/Beacon/getBlockHeaders

**Which issues(s) does this PR fix?**

Part of https://github.com/prysmaticlabs/prysm/issues/12635

**Other notes for review**

Tested on checkpoint sync node
Head slot: `0.003s`
Older slot s1 (25 min ago): `0.003s`
Parent root for slot s1: `0.003s`
Significantly faster than lighthouse (10x) on my machine

### CODE REVIEW

**Comment by anukul** [comment]
cc @rkapka

File: beacon-chain/rpc/eth/beacon/handlers.go [hunk 1]
```diff
@@ -640,6 +641,132 @@ func (bs *Server) validateEquivocation(blk interfaces.ReadOnlyBeaconBlock) error
 	return nil
 }
 
+// ListBlockHeaders retrieves block headers matching given query. By default it will fetch current head slot blocks.
+func (bs *Server) ListBlockHeaders(w http.ResponseWriter, r *http.Request) {
```
Comment (rkapka) [inline beacon-chain/rpc/eth/beacon/handlers.go:645]: Let's follow the convention of naming the receiver `s` in HTTP handlers and rename the function to `GetBlockHeaders`.

File: beacon-chain/rpc/eth/beacon/handlers.go [hunk 2]
```diff
@@ -640,6 +641,132 @@ func (bs *Server) validateEquivocation(blk interfaces.ReadOnlyBeaconBlock) error
...
+	var blks []interfaces.ReadOnlySignedBeaconBlock
+	var blkRoots [][32]byte
+
+	if rawParentRoot != "" {
+		if valid := shared.ValidateHex(w, "Parent Root", rawParentRoot); !valid {
+			return
+		}
+		parentRoot, err := hexutil.Decode(rawParentRoot)
+		if err != nil {
+			errJson := &http2.DefaultErrorJson{
```
Comment (rkapka) [inline beacon-chain/rpc/eth/beacon/handlers.go:662]: We have a `HandleError` method that will make the code more concise. Same comment for other usages of the error struct.

File: beacon-chain/rpc/eth/beacon/structs.go [hunk 3]
```diff
@@ -233,6 +233,18 @@ type VoluntaryExit struct {
 	ValidatorIndex string `json:"validator_index" validate:"required"`
 }
 
+type ListBlockHeadersResponse struct {
```
Comment (rkapka) [inline beacon-chain/rpc/eth/beacon/structs.go:236]: Let's rename to `GetBlockHeadersResponse`

**Review by rkapka — APPROVED** [review]

**Pull request merged**